python/plugins/MetaSearch/util.py

```diff
@@ -167,3 +166,10 @@ def clean_ows_url(url):
         url.setQuery(query_string)
 
     return url.toString()
+
+
+def log_message(message, level):
+    """helper function to emit logging messages"""
+
+    message_translated = QCoreApplication.translate('MetaSearch', message)
+    LOGGER.logMessage(message_translated, 'MetaSearch', getattr(Qgis, level))
```

Comment:
```suggestion
    LOGGER.logMessage(message_translated, 'MetaSearch', getattr(Qgis, level.capitalize(), 'Info'))
```

Hi @tomkralidis, I would not use a string (e.g. `Info`, `Warn` etc.) to specify the level, but use dedicated methods (`log_info`, `log_warning` etc.) instead, but maybe that's overkill.
You could also make sure that the `level` has the correct capitalization, and - if the attribute is not found - set a default log level?

Anyway, thanks for adding some logging! It helps to see what MetaSearch is doing under the hood :)

Reply:
Thanks @GeoSander.  I've pushed an update just now that adds a default (`Info`).

Reply:
Why not just use Qgis.MessageLevel directly here instead? The current approach with passing a string and then resolving via getattr looks needlessly fragile.

Reply:
Updated.